DeratControl.Domain/Entities/Facility.cs

```diff
@@ -17,5 +18,16 @@ private Facility()
             this.Perimeters = new List<Perimeter>();
             this.Reviews = new List<Review>();
         }
+        public void AddPerimeter(PerimeterType perimeterType, User user)
+        {
+            foreach (var perimeter in this.Perimeters)
```

Comment:
var perimeterExists = this.Perimeters.Any(c => c.PerimeterType == perimeterType);